src/components/plotly/shared/plot.js

```diff
@@ -5,34 +5,45 @@ import Plotly from 'plotly.js-basic-dist-min'
 
 import Styles from './styles'
 
+
 // credit JamesRamm
 // https://github.com/plotly/react-plotly.js/issues/242
-const Plot = ({ data, layout, config }) => {
+const Plot = ({ data, layout, config, onAfterPlot }) => {
   const [ref, setRef] = useState(null)
+
   useLayoutEffect(() => {
     try {
       if (data && layout && config) {
-        ref && Plotly.react(ref, { data, layout, config }).catch(() => { })
+        ref && Plotly.react(ref, { data, layout, config })
+          .then((res) => { 
+            onAfterPlot({ response: res, isLoaded: true })
+          })
+          .catch((err) => {
+            onAfterPlot({ response: err, isLoaded: false })
+          })
```

Comment:
Not new with this change, but how does this work with promise and synchronous try/catch together? Seems a bit weird and not as coherent

Also prefer using `if` than `&&` to carry out the logic execution (unless in JSX context)

Reply:
It is a promise from plotly side which returns the Ref of the graph when it is loaded in the DOM.

Reply:
Agree with Leo, include ref into the above if conditions.